library/ui/src/main/java/com/google/android/exoplayer2/ui/PlayerView.java

```diff
@@ -1055,21 +1071,30 @@ private boolean setArtworkFromMetadata(Metadata metadata) {
       if (metadataEntry instanceof ApicFrame) {
         byte[] bitmapData = ((ApicFrame) metadataEntry).pictureData;
         Bitmap bitmap = BitmapFactory.decodeByteArray(bitmapData, 0, bitmapData.length);
-        return setArtworkFromBitmap(bitmap);
+        return setArtworkFromBitmap(new BitmapDrawable(getResources(), bitmap));
       }
     }
     return false;
   }
 
-  private boolean setArtworkFromBitmap(Bitmap bitmap) {
-    if (bitmap != null) {
-      int bitmapWidth = bitmap.getWidth();
-      int bitmapHeight = bitmap.getHeight();
+  private boolean setDrawableArtwork(Drawable drawable) {
+    if(drawable instanceof BitmapDrawable) {
```

Comment:
Is this code path necessary? I was thinking you could just pass the `BitmapDrawable` drawable to `setImageDrawable` and the right thing would happen, rather than having to special case this to get the bitmap back out again.

Reply:
Perhaps this is related to calling `contentFrame.setAspectRatio`. Although looking at that, isn't it problematic to be calling it for the bitmap case only? Setting a bitmap will cause the aspect ratio to become to a particular value, which will then still be applied should another type of `Drawable` be set later on. Do `getIntrinsic[Width/Height]` on a `Drawable` do anything useful for this (I've no idea)?

I think one way or another we need to get to a single code path for this to fall out nicely; I'm just not quite sure how to do that.

Reply:
Right, I mainly left it because of the aspect ratio, but as you said, calling `setImageDrawable()` should do it, and since the `artWork` container is an `ImageView`, the user has the flexibility to apply his own scale type and/or `adjustViewBounds`. So I think we can remove that logic from here. What do you think?

Reply:
I think that sounds right. If the content frame aspect ratio is unnecessary then you probably want to explicitly set it to 0, in case it was set previously (e.g. due to a video being played).

Reply:
Ok, just updated, let's check if that makes more sense. Thanks a lot!